src/services/grpc/api/login.ts

```diff
@@ -16,74 +15,43 @@ export async function login(request: LoginRequest): Promise<DeepPartial<LoginRes
 		throw new ServerError(Status.INVALID_ARGUMENT, 'Invalid grant type');
 	}
 
-	if (grantType === 'password' && !username) {
-		throw new ServerError(Status.INVALID_ARGUMENT, 'Invalid or missing username');
-	}
-
-	if (grantType === 'password' && !password) {
-		throw new ServerError(Status.INVALID_ARGUMENT, 'Invalid or missing password');
-	}
-
-	if (grantType === 'refresh_token' && !refreshToken) {
-		throw new ServerError(Status.INVALID_ARGUMENT, 'Invalid or missing refresh token');
-	}
-
 	let pnid: HydratedPNIDDocument | null;
 
 	if (grantType === 'password') {
-		pnid = await getPNIDByUsername(username!); // * We know username will never be null here
+		if (!username) throw new ServerError(Status.INVALID_ARGUMENT, 'Invalid or missing username');
+		if (!password) throw new ServerError(Status.INVALID_ARGUMENT, 'Invalid or missing password');
 
-		if (!pnid) {
-			throw new ServerError(Status.INVALID_ARGUMENT, 'User not found');
-		}
+		pnid = await getPNIDByUsername(username); // * We know username will never be null here
```

Comment:
This comment is useless here, it only existed because we were using `!` and TypeScript was throwing a fit